### PR TITLE
[embedded] In -no-allocations, avoid linking allocating/deallocating runtime functions

### PR DESCRIPTION
**Pull request by kubamracek**
In -no-allocations, avoid linking allocating/deallocating runtime functions. Also add an explicit dependency test.

### CODE REVIEW

File: include/swift/AST/DiagnosticsSIL.def [hunk 1]
```diff
@@ -364,6 +364,10 @@ ERROR(embedded_swift_metatype_type,none,
       "cannot use metatype of type %0 in embedded Swift", (Type))
 ERROR(embedded_swift_metatype,none,
       "cannot use metatype in embedded Swift", ())
+ERROR(embedded_swift_allocating_type,none,
+      "cannot use allocating type %0 in -no-allocations mode", (Type))
+ERROR(embedded_swift_allocating,none,
+      "cannot use alllocating type in -no-allocations mode", ())
```
Comment (rauhul) [inline include/swift/AST/DiagnosticsSIL.def:370]: You have an accidental extra "l" in "allocating"

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by eeckstein — APPROVED** [review]
lgtm

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test macOS platform

**Pull request merged**